overlord/devicestate/firstboot.go

```diff
@@ -339,7 +339,7 @@ func importAssertionsFromSeed(st *state.State, sysLabel string, hasModeEnv bool)
 	// collect and
 	// set device,model from the model assertion
 	deviceSeed, err := loadDeviceSeed(st, sysLabel)
-	if err == seed.ErrNoAssertions && !hasModeEnv && release.OnClassic {
+	if err == seed.ErrNoAssertions && !isCoreBoot && release.OnClassic {
```

Comment:
with the correctly defined isCoreBoot we don't need the `&& release.OnClassic` part anymore